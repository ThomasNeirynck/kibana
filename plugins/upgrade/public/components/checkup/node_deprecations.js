import _ from 'lodash';
import PropTypes from 'prop-types';
import React from 'react';

import { StatusGroup } from '../status_group';
import { Issues } from './issues';
import { IssueSummary } from './issue_summary';


export function NodeDeprecations({ className, deprecations }) {
  const issueLevelCounts = _.countBy(deprecations, 'level');

  return (
    <StatusGroup
      className={ className }
      isInitiallyCollapsed
      status={ <IssueSummary issueLevelCounts={ issueLevelCounts } /> }
      title="Node Settings"
    >
      { deprecations.length > 0
        ? <Issues issues={ deprecations } />
        : <p className="kuiText kuiSubduedText">No node settings deprecations</p>
      }
    </StatusGroup>
  );
}

NodeDeprecations.propTypes = {
  className: PropTypes.string,
  deprecations: PropTypes.array,
};

NodeDeprecations.defaultProps = {
  className: null,
  deprecations: [],
};
