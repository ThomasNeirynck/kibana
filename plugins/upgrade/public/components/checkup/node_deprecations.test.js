import { shallow } from 'enzyme';
import React from 'react';

import { NodeDeprecations } from './node_deprecations';

jest.mock('ui/chrome', () => {}, { virtual: true });
jest.mock('ui/notify/notifier', () => ({
  Notifier: function () {},
}), { virtual: true });


describe('NodeDeprecations', () => {
  test('renders the className property', () => {
    const component = (
      <NodeDeprecations className="myClass" />
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('renders a success message when there are no deprecations', () => {
    const component = (
      <NodeDeprecations deprecations={[]}/>
    );

    expect(shallow(component)).toMatchSnapshot();
  });

  test('renders Issues and IssueSummary components when there are deprecations', () => {
    const deprecations = [
      { level: 'info', },
      { level: 'critical', },
      { level: 'critical', },
    ];
    const component = (
      <NodeDeprecations deprecations={deprecations}/>
    );

    expect(shallow(component)).toMatchSnapshot();
  });
});
