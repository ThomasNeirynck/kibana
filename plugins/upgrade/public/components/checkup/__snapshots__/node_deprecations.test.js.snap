// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeDeprecations renders Issues and IssueSummary components when there are deprecations 1`] = `
<StatusGroup
  className={null}
  isInitiallyCollapsed={true}
  status={
    <IssueSummary
      className={null}
      issueLevelCounts={
        Object {
          "critical": 2,
          "info": 1,
        }
      }
    />
  }
  title="Node Settings"
>
  <Issues
    className={null}
    issues={
      Array [
        Object {
          "level": "info",
        },
        Object {
          "level": "critical",
        },
        Object {
          "level": "critical",
        },
      ]
    }
  />
</StatusGroup>
`;

exports[`NodeDeprecations renders a success message when there are no deprecations 1`] = `
<StatusGroup
  className={null}
  isInitiallyCollapsed={true}
  status={
    <IssueSummary
      className={null}
      issueLevelCounts={Object {}}
    />
  }
  title="Node Settings"
>
  <p
    className="kuiText kuiSubduedText"
  >
    No node settings deprecations
  </p>
</StatusGroup>
`;

exports[`NodeDeprecations renders the className property 1`] = `
<StatusGroup
  className="myClass"
  isInitiallyCollapsed={true}
  status={
    <IssueSummary
      className={null}
      issueLevelCounts={Object {}}
    />
  }
  title="Node Settings"
>
  <p
    className="kuiText kuiSubduedText"
  >
    No node settings deprecations
  </p>
</StatusGroup>
`;
